WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1068 @@
+<?php
+/**
+ * Represents a PHP_CodeSniffer sniff for sniffing WordPress coding standards.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ */
+
+/**
+ * Represents a PHP_CodeSniffer sniff for sniffing WordPress coding standards.
+ *
+ * Provides a bootstrap for the sniffs, to reduce code duplication.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Juliette Reinders Folmer <wpplugins_nospam@adviesenzo.nl>
+ * @author    Kevin Haig
+ * @author    Ulrich Pogson <ulrich@pogson.ch>
+ */
+abstract class WordPress_AbstractThemeSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Array holding the once through WordPress Theme file checks.
+	 *
+	 * @var array
+	 */
+	private static $sniff_helper;
+
+	/**
+	 * Initial setup.
+	 *
+	 * @var array $theme_data contains data from style.css headers.
+	 */
+	public function __construct() {
+		// do a once through pass and get the files.
+		if ( empty( self::$sniff_helper ) || ! is_array( self::$sniff_helper ) || false === self::$sniff_helper ) {
+			$files = $this->get_files();
+			$themedir = $files[0];
+			// Only run this if checking a directory.
+			$themedir_parts = pathinfo( $themedir );
+			if ( isset( $themedir_parts['extension'] ) ) {
+				return;
+			}
+			$themefiles = $this->listdir( $themedir );
+			self::$sniff_helper = $this->once_through( $themedir , $themefiles );
+			// var_dump( self::$sniff_helper );.
+		}
+	}
+
+	/**
+	 * Get the list of files that are being sniffed.
+	 *
+	 * @return array
+	 */
+	private function get_files() {
+		$cli_object = new PHP_CodeSniffer_CLI;
+		$command_line_values = $cli_object->getCommandLineValues();
+		array_multisort( array_map( 'strlen', $command_line_values['files'] ), $command_line_values['files'] );
+		return $command_line_values['files'];
+	}
+
+	// ================== below are functions for the once through
+	/**
+	 * Taken from themecheck
```

Comment:
> As the code was originally written by @Pross We can ask him if he would allow us to licence the code under MIT?

FYI: AFAIK for a change of license you have to get permission from everybody who has contributed to the plugin as it is at the time of the license change.